src/ray/object_manager/plasma/store.h

```diff
@@ -48,13 +49,98 @@ using flatbuf::PlasmaError;
 
 struct GetRequest;
 
+using CreateObjectCallback = std::function<Status(const std::shared_ptr<Client> &client, bool reply_on_oom, bool evict_if_full)>;
+
+class CreateRequestQueue {
+ public:
+  CreateRequestQueue(int32_t max_retries,
+      uint32_t delay_on_oom_ms,
+      uint32_t delay_on_transient_oom_ms,
+      bool evict_if_full,
+      std::function<void()> on_store_full)
+    : max_retries_(max_retries),
+      delay_on_oom_ms_(delay_on_oom_ms),
+      delay_on_transient_oom_ms_(delay_on_transient_oom_ms),
+      evict_if_full_(evict_if_full),
+      on_store_full_(on_store_full) {}
+
+  /// Add a request to the queue.
+  ///
+  /// The request may not get tried immediately if the head of the queue is not
+  /// serviceable.
+  ///
+  /// \param client The client that sent the request.
+  void AddRequest(const std::shared_ptr<Client> &client, const CreateObjectCallback &request_callback);
+
+  /// Process requests in the queue.
+  ///
+  /// This will try to process as many requests in the queue as possible, in
+  /// FIFO order. If the first request is not serviceable, this will break and
+  /// the caller should try again later.
+  ///
+  /// \param[out] retry_after_ms The amount of time after which the first
```

Comment:
Can we remove this and just have a small internal retry timeout of say `10ms`? Seems harmless to fire a timer at that frequency.

Reply:
See my comment on the main PR thread about why I have two timeouts.

I'll set the internal one to 10ms, but let me know what you think about having the two timeouts. I worry about using such a low timeout for the intransient OOM error.